app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,31 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @posts = Post.where(user_id: params[:user_id]).order(created_at: :desc)
+    @user = User.find_by(id: params[:user_id])
+  end
 
-  def show; end
+  def show
+    @post = Post.find_by(id: params[:id])
+    @comments = Comment.where(post_id: params[:id]).order(created_at: :desc)
+  end
+
+  def create_post
+    @post = Post.new
+    @post.user_id = current_user.id
+  end
+
+  def create
+    @post = Post.new
+    @post.likes_counter = 0
+    @post.comments_counter = 0
+    user = current_user
+    @post.user_id = user.id
+    temp = params[:post]
+    text = temp[:text]
+    title = temp[:title]
+    @post.text = text
+    @post.title = title
+
+    redirect_to("/users/#{user.id}/posts/#{@post.id}") if @post.save
```

Comment:
* Kindly add strong parameters here for example like this ▶️ `  
* `private `
   ` def post_params `
     ` params.require(:post).permit(:title, :text)`
   ` end`
* As This is a better security practice to help prevent accidentally allowing users to update sensitive model attributes.  [Refer back to lesson link](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides.)